streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -800,6 +804,9 @@ record = null;
             record = null;
             throw exception;
         } catch (final RuntimeException e) {
+            // Do not keep the failed processing exception in the stack trace
+            final Throwable processingException = e instanceof FailedProcessingException ? e.getCause() : e;
```

Comment:
We should not use `instanceof` but add a new `catch (FailedProcessingException)` block to cover this case.

Reply:
Having a `catch (FailedProcessingException)` will be now required anyway since `FailedProcessingException` will now extend `StreamsException`, and we don't `FailedProcessingException` to be caught by the `catch (StreamsException)` block from above.